### DISCUSSION TITLE
Api get content of file for organization repository

### DISCUSSION THREAD

**Discussion by mapayares** [community/community#24906] [Apps, API and Webhooks]
<p>I am writing a java program that retrieves file contents from our github repository. In reading github documentaion to retrieve a file i must hit this github endpoint </p>
<pre><code>GET /repos/:owner/:repo/contents/:path 
</code></pre>
<p> the question I have is what to substitute for <strong>owner</strong>.  Let’s say my <strong>personal</strong> github user is <strong>mike</strong>  but the owner of the github repository is <strong>acme</strong>. The first GET path pasted below works in my program. I am able to retrieve the file content with no issues. If I try to substitute <strong>mike</strong> for <strong> acme</strong> the path below does not work. I understand why it is not successful because that url path does not exist in github. Does that mean the API token I generate for <strong>mike</strong>  cannot be used to retrieve the content in the repository owned by <strong>acme</strong>? Or can I retrieve content using the  <strong>mike</strong> api token even though <strong>mike</strong>  is not the owner for the <strong>acme</strong> repository (but has access to it)? The repository in question " <strong>java-repo"</strong>  that has many contributors. Does this mean there can only be one user that accesses the repository in this case <strong>acme </strong> via the api’s?</p>
<p><strong>GET /repost/acme/java-repo/contents/file-path.java – works</strong></p>
<p><strong>GET /repost/mike/java-repo/contents/file-path.java – does not work</strong></p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/mapayares">@mapayares</a>,</p>
<p>Thanks for being part of the GitHub Community Forum! I’ll do my best to help. </p>
<p>The</p>
<pre><code>:owner
</code></pre>
<p>is always going to be the individual or organization that actually owns the repository in question. The second URL you shared doesn’t work, as you noted, simply because the content doesn’t exist on your <code>mike</code> account. The URL is essentially invalid.</p>
<p>Additionally, API tokens are scoped to your user account. So, if your user account has only read access to the <code>acme</code> repository, then your API token can be granted at most read only access by applying the correct scopes. So, at most, the program can only do things that your <code>mike</code> account can do, but you can also restrict the program to fewer abilities, depending on how you generate the token.</p>
<p>I hope this helps! </p>

**Comment by mapayares** [comment]
<p>thank you so much for your help. i was able to confirmed that it works as you described. thanks again</p>
